.github/workflows/build_test.sh

```diff
@@ -130,17 +130,22 @@ ninja --version
 
 for args in "${ARGS[@]}"; do
     SECONDS=0
+    FCOMMON="-Werror"
 
     if [[ "$COMPILER" == clang && "$args" =~ Wmaybe-uninitialized ]]; then
         # -Wmaybe-uninitialized is not implemented in clang
         continue
     fi
 
+    if [[ "${CRYPTOLIB:?}" != openssl ]]; then
+        FCOMMON="$FCOMMON -DOPENSSL_NO_DEPRECATED"
```

Comment:
It's fine to add a single job for this, but not to disable used APIs in all jobs, as we lose coverage of them. So change this to only affect a single job instead.

Reply:
outside the "engines" no used apis are disabled. mostly stuff is changed between openssl 1.x and 3.x, so a test with 1.x would make most sense?

theoretical we could also dlsym the engine stuff, at which point nothing should change for unit tests.

Reply:
The engines interface is exactly what we need coverage for

Reply:
Yes. @bluca is right. Please introduce a new test job with the flag.

Reply:
This is still unresolved. Engines with Openssl3 must be covered by CIs, as it's the only realistic way to use pkcs11 and other hardware stuff with our tools.